File: frontend/src/types/room.ts
export interface SceneToken {
    id: string;
    name: string;
    x: number; // percentage 0-100
    y: number; // percentage 0-100
    color?: string;
    avatar?: string;
    character_id?: number; // ID do personagem vinculado
    initiative?: number; // Valor de iniciativa para ordem de combate
    current_hp?: number; // HP atual (override do personagem)
}

export interface SceneState {
    backgroundUrl?: string;
    tokens: SceneToken[];
    notes?: string;
    [key: string]: any;
}

export interface RoomChatMessage {
    id: string;
    room_id: string;
    user_id: number;
    message: string;
    timestamp: number;
}

export interface RoomMember {
    room_id: string;
    user_id: number;
    role: string;
    joined_at: string;
}

export interface Room {
    id: string;
    name: string;
    owner_id: number;
    campaign_id?: number;
    scene_state?: SceneState;
    created_at: string;
    updated_at: string;
    members?: RoomMember[];
    metadata?: Record<string, any>;
}

export interface RoomSocketEvent {
    type: string;
    room_id?: string;
    sender_id?: number;
    message?: string;
    scene_state?: SceneState;
    dice?: any;
    metadata?: Record<string, any>;
    members?: number[];
    timestamp?: number;
}

File: backend/internal/models/room.go
package models

import "time"

// Room represents a collaborative play space (MVP, stored in-memory for now).
type Room struct {
	ID         string        `json:"id" db:"id"`
	Name       string        `json:"name" db:"name"`
	OwnerID    int           `json:"owner_id" db:"owner_id"`
	CampaignID *int          `json:"campaign_id,omitempty" db:"campaign_id"`
	SceneState JSONBFlexible `json:"scene_state,omitempty" db:"scene_state"`
	CreatedAt  time.Time     `json:"created_at" db:"created_at"`
	UpdatedAt  time.Time     `json:"updated_at" db:"updated_at"`
	Members    []RoomMember  `json:"members,omitempty"`
	Metadata   JSONB         `json:"metadata,omitempty" db:"metadata"`
}

// RoomMember links a user to a room with a role.
type RoomMember struct {
	RoomID   string    `json:"room_id" db:"room_id"`
	UserID   int       `json:"user_id" db:"user_id"`
	Role     string    `json:"role" db:"role"` // e.g. "gm" or "player"
	JoinedAt time.Time `json:"joined_at" db:"joined_at"`
}
